### DISCUSSION TITLE
Getting text rendering mode

### DISCUSSION THREAD

**Discussion by szattila98** [jsvine/pdfplumber#480] [Q&A] [answered]
Hi, I would like to use this tool to extract text lines from a pdf. I need the text lines themselves, the position of the textlines, the position and dimensions of images and the rendering modes of text lines (it is needed to indentify invisible or non stroke/fill text). The first three should be possible if I understood the readme correctly, but I don't see anything about the rendering mode. Could someone please tell me whether it is possible to somehow get the rendering modes of text lines or characters? Thanks in advance!

**Comment by jsvine** [comment, marked as answer]
Hi @szattila98, and thanks for your interest. You are correct: Getting the text-rendering mode is currently not possible through `pdfplumber`, largely because `pdfminer.six` does not expose this information through its core rendering objects. If that changes, it would indeed be great to incorporate that information into `pdfplumber`, too.

**Reply by szattila98** [reply]
Thank you for the reply and the info.
